artpaint/application/UndoQueue.cpp

```diff
@@ -371,9 +372,9 @@ void UndoQueue::UpdateMenuItems()
 	event_name = ReturnRedoEventName();
 	if (redo_menu_item != NULL) {
 		if (event_name != NULL) {
-			char menu_text[256];
-			sprintf(menu_text,"%s %s",B_TRANSLATE("Redo"),event_name);
-			redo_menu_item->SetLabel(menu_text);
+			BString menu_text;
+			menu_text.SetToFormat(B_TRANSLATE("Redo %s"), event_name);
```

Comment:
also is there a reason you didn't do the same treatment for Redo?

Reply:
Frankly I didn't have time to open my Haiku VM, just applied @humdingerb's suggestion 😂. I will return to this in a day or two.

Reply:
Please don't push untested commits.  I @humdingerb catches a lot of my bugs but I always try to test them first.